lib/parser.mly

```diff
@@ -55,15 +54,25 @@ opt_params_list:
 
 /* define foo(number bar -> string) */
 fdecl:
-  DEFINE ID LPAREN opt_params_list GIVES func_rtype RPAREN COLON EOL INDENT stmt_list DEDENT
-  {
+    DEFINE ID LPAREN opt_params_list GIVES func_rtype RPAREN COLON EOL INDENT stmt_list DEDENT
     {
-      fname=$2;
-      params=$4;
-      rtype=$6;
-      body=$11
+      {
+        fname=$2;
+        params=$4;
+        rtype=$6;
+        body=$11
+      }
     }
-  }
+  | stmt_list EOF 
```

Comment:
a shift-reduce conflict might come from here? I think we allow for stmt lists somewhere else. Maybe we could make the `program` rule at the top into this:

`program: bind_list fdecl_list stmt_list`

This way we don't need two different ways for defining a function.

Reply:
so I had this at first, but then you had to combine the `stmt_list` into `main` in the semant or elsewhere. iirc that was causing some other issues. btw there are not any shift/reduce conflicts anymore, just parse errors :(

Reply:
:(